packages/core/src/utils.ts

```diff
@@ -381,13 +381,13 @@ export function toObserver<T>(
 }
 
 export function createInvokeId(stateNodeId: string, index: number): string {
-  return `${stateNodeId}[${index}]`;
+  return `${index}.#${stateNodeId}`;
 }
 
 export function resolveReferencedActor(machine: AnyStateMachine, src: string) {
-  if (src.startsWith('xstate#')) {
-    const [, indexStr] = src.match(/\[(\d+)\]$/)!;
-    const node = machine.getStateNodeById(src.slice(7, -(indexStr.length + 2)));
+  if (src.startsWith('xstate.source')) {
```

Comment:
I'd expect this to be enough already:
```suggestion
  if (src.startsWith('xstate.')) {
```

Reply:
kinda addressed this in a different way in https://github.com/statelyai/xstate/pull/4524